### PR TITLE
Improve: let-module construct 

### PR DESCRIPTION
**Pull request by hhugo**

### CODE REVIEW

File: test/passing/skip.ml [hunk 1]
```diff
@@ -93,14 +93,16 @@ let _ = (x [@ocamlformat "disable"]  [@test? _ when e [@test 2]]) 3
 let _ =
   let module X = struct
     let x = 4
-  end in
+  end
+  in
```
Comment (jberdine) [inline test/passing/skip.ml]: Do you consider this an improvement? Or is this change to recover a bit from the effects of `disable` in the cases below?

Comment (hhugo) [reply test/passing/skip.ml]: It looks more consistent with other `let-in` where `in` fall on its own line in case of a break.

Comment (hhugo) [reply test/passing/skip.ml]: To be clear, the change about `let module M = struct end in` was not the target of that PR. Maybe I should restrict the scope and special case to remove this side-effect.

Comment (jberdine) [reply test/passing/skip.ml]: Ok, then I am not sure of the intended change, can you point to an example?

Comment (hhugo) [reply test/passing/skip.ml]: well, the regtest `module.ml` that was removed from failing and that I forgot add into passing :)

File: test/passing/source.ml.ref [hunk 2]
```diff
@@ -3464,7 +3472,8 @@ type ('k, 'd, 'm) map =
 let add (type k d m) (m : (k, d, m) map) x y s =
   let module M = ( val m
                      : MapT
-                     with type key = k and type data = d and type map = m ) in
+                     with type key = k and type data = d and type map = m )
+  in
```
Comment (hhugo) [inline test/passing/source.ml.ref]: this is what this PR targets

File: test/passing/skip.ml (hunk 1 above)
Comment (jberdine) [reply test/passing/skip.ml]: Ok, let me look

**Comment by hhugo** [comment]
I've added a special case for Pmod_structure

**Comment by jberdine** [comment]
What do you think of #328 instead?

**Comment by hhugo** [comment]
lgtm

**Pull request closed without merging**